packages/kbn-management/settings/components/field_row/input_footer/reset_link.tsx

```diff
@@ -30,9 +36,21 @@ export const DATA_TEST_SUBJ_RESET_PREFIX = 'management-settings-resetField';
  */
 export const FieldResetLink = <T extends SettingType>({
   onReset,
+  unsavedChange,
   field,
 }: FieldResetLinkProps<T>) => {
-  if (field.isDefaultValue) {
+  // If there's no unsaved change, and the field saved value is the default value, don't
```

Comment:
When I make the default value of the image setting the same as the saved value, I notice some issues. The "Change link" doesn't hide, and when I select some image and then remove it, the "Revert to default" link incorrectly shows up and doesn't do anything when clicked. I actually think that this behavior exists before the changes from this PR, but I didn't notice it before.

https://github.com/elastic/kibana/assets/59341489/20da7b6c-3c50-48c2-97a6-dda3537159c2

Reply:
I'll look into this and fix before shipping... thanks!

Reply:
This actually shouldn't be available in Storybook stories, because the code in `advancedSettings`-- and thus, here-- assumes an image-based setting will never have a default value other than null.

I'm now tracking this here: https://github.com/elastic/kibana/issues/166578